scala/core/src/sprites/ACSetEntitySources.scala

```diff
@@ -79,3 +85,53 @@ def ACSetEdgeSource(
     tgt: Hom,
     sprite: Sprite
 ) = ACSetEntitySource(ob, sprite).addPropsBy(edgeProps(src, tgt))
+
+/** Similar to [[edgeProps]]. Computes the position and direction for the ends
+  * of a wire from the ports it is connected to, using the top-level properties
+  * in `acs` and the other sprites in `m`.
+  *
+  * If `src`/`tgt` are present, it uses those, otherwise it looks up an explicit
+  * `Start`/`End` value in `acs`
+  *
+  * Takes an optional callback argument `bg: => Part` to relativize the lookup
+  * to a variable background for zooming in and out.
+  */
+
+def wireProps(
+    src: Hom,
+    tgt: Hom,
+    typeProps: (ACSet, Part) => PropMap,
+    dir: Part => Complex = _ => Complex(0, 0),
+    bg: => Part = ROOT
+)(_e: Entity, acs: ACSet, m: EntityMap): PropMap = {
+
+  val p = acs.props
+
+  val Seq(s, t) = Seq(src, tgt).map(p.get(_))
+
+  val sc = s
+    .flatMap(_.diffOption(bg))
+    .flatMap(findCenter(_, m))
+    .getOrElse(
+      p.get(Start).getOrElse {
+        Complex(500, 500)
```

Comment:
This hardcoded value is big code smell... Is there another way to do this?

Reply:
I've added an explicit `defaultPos` val to clean it up a little. Unfortunately, removing the default turns out to be a big headache.

For some reason I'm having trouble tracking down, modifying the ACSet initiates a `present` operation on all the parts in `ROOT`, even when they are not visible in the viewport. Happy to talk more about how to fix this.

Reply:
Wait, that doesn't make sense at all; present should only run once per part *ever*.